YAKL_parallel_for_c.h

```diff
@@ -572,6 +572,8 @@ namespace c {
   #ifdef __USE_SYCL__
     template<class F, int N, bool simple>
     void parallel_for_sycl( Bounds<N,simple> const &bounds , F const &f , int vectorSize = 128 ) {
+      isTriviallyCopyable<F>();
```

Comment:
Just requiring is_trivially_copyable.

In the future, this will become `is_memcpyable`